main.c

```diff
@@ -1,21 +1,23 @@
-#include "list.h"
+#include <stdio.h>
 
-int main()
+#include "binary_tree.h"
+
+int main(int argc, char *argv[], char *envp[])
 {
-    struct slist head;
-    struct slist *headp = &head;
-    
-    INIT_NODE(head, 10);
-    print_all_slist(headp);
-    add_node_beg(&headp, create_node(20));
-    add_node_beg(&headp, create_node(30));
-    add_node_beg(&headp, create_node(40));
-    print_all_slist(headp);
-    add_node_end(headp, create_node(9));
-    add_node_end(headp, create_node(4));
-    print_all_slist(headp);
-    add_node_pos(&headp, create_node(5), 5);
-    add_node_pos(&headp, create_node(1), 1);
-    add_node_pos(&headp, create_node(200), 200);
-    print_all_slist(headp);
+    // Note: Previous Linked-List feedback applied here - root is now on heap
+    struct bt_node *root = bt_new_node(10);
```

Comment:
Thanks for pointing that out, latest push now does assert checks and uses strerror() to print reason for memory allocation failure.